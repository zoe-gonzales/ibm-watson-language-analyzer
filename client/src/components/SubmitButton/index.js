import React from 'react';

const SubmitButton = ({ onClick }) => {
    return (
        <button
            className="btn waves-effect waves-light"
            type="submit" 
            name="action"
            onClick={onClick}>
            <i className="material-icons right">send</i>
        </button>
    )
}

export default SubmitButton;