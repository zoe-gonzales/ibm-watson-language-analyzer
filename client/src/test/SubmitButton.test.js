import React from 'react';
import SubmitButton from '../components/SubmitButton';
import renderer from 'react-test-renderer';

it('renders SubmitButton component correctly', () => {
    const handleClick = e => {
        console.log(`Event: ${e.target}`);
        console.log('Click event recognized successfully');
    };
    const component = renderer
    .create(<SubmitButton onClick={handleClick} />)
    .toJSON();
    expect(component).toMatchSnapshot();
});