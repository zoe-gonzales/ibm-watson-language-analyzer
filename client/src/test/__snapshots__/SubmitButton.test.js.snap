// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SubmitButton component correctly 1`] = `
<button
  className="btn waves-effect waves-light"
  name="action"
  onClick={[Function]}
  type="submit"
>
  <i
    className="material-icons right"
  >
    send
  </i>
</button>
`;
